content/blog/2022/shared-elements.md

```diff
@@ -0,0 +1,420 @@
+the changes in an element from state 'A' to state 'B'.
+Again, we're describing the common use-case
+for a FLIP animation.
+
+## Let's try it out!
+
+For my own talk at SmashingConf,
+I created a
+[visualization](https://slides.oddbird.net/demos/cascade/layers/)
+of
+[Cascade Layers](https://css-tricks.com/css-cascade-layers/)
+being 'sorted' into groups.
+The visualization starts
+with an explicit layer order
+followed by 9 layer blocks & imports,
+mixed together in random order.
+Then three buttons allowing them
+to be shown either in their original order,
+sorted so last-takes-precedence,
+or weighted with most powerful at the top:
+
+{{ embed.figure(
+  data=layers.gallery,
+  caption=layers.caption
+) }}
+
+Pressing a given button
+changes a `data-sort` attribute
+on the (flexbox) list of layers,
+and then CSS `order` and `flex-direction`
+(both discreet values)
+are used to re-order the list
+in different ways.
+I wanted to animate the re-ordering,
+to make it clear what's happening,
+but in a hurry
+I didn't want to mess with
+the JavaScript libraries required.
+
+It may also be more semantically appropriate
+to do this sorting in the DOM --
+and I may still do that --
+but at the time I was more concerned
+about the resulting stage visualization.
+Either way,
+sorting in CSS or in the DOM,
+the shared-element transition will work.
+All that matters to this API
+is that _something changed_ between two states,
+and we can animate each 'shared element'
+from one state to the other.
+
+In this case, we want to associate
+each element in the initial state
+with the same element in the target state.
+Since the element itself isn't going anywhere,
+we can do that by giving each element
+a unique `page-transition-tag`.
+That transition-tag will be the same
+both before and after the state-change,
+associating the element before
+with the same element after.
+I used a `for` loop in Sass
+to apply unique transition-tags on each layer:
+
+```scss
+// for each of the nine layers blocks
+@for $i from 1 through 9 {
+  // give it a transition-tag based on its nth-position
+  [data-layer]:nth-of-type(#{$i}) {
+    page-transition-tag: layer-#{$i};
+  }
+}
+```
+
+That's all we need in the CSS.
+I also have a `pressBtn` JavaScript function
+that changes an attribute on the list.
+This is the function that 'triggers' our change:
+
+```js
+const btnPress = (btn) => {
+  list.setAttribute('data-sort', btn.dataset.set);
+}
+```
+
+All that's left is telling the browser
+that this attribute change
+should be considered a 'document transition'.
+As far as I can tell,
+with my limited research,
+this seems to work:
+
+```js
+const btnPress = (btn) => {
+  // create a document transition
+  const transition = document.createDocumentTransition();
+
+  // start the transition,
+  // and make our change in the callback function
+  await transition.start(() =>
+    doc.setAttribute("data-sort", btn.dataset.set)
+  );
+}
+```
+
+Roughly,
```

Comment:
I find this wording a bit confusing, but I'm trying to figure out a better suggestion...

```suggestion
Roughly speaking,
```